### PR TITLE
[2.0.2] *: add `transfer-source` command

### PR DESCRIPTION
**Pull request by lance6716**
### What is changed, added, or deleted? (Required)

align with https://github.com/pingcap/dm/pull/1492

### Which DM version(s) do your changes apply to? (Required)

- [x] master (the latest development version, including v3.0 changes)
- [x] v2.0 (TiDB DM 2.0 versions)
- [ ] v1.0 (TiDB DM 1.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: zh/manage-source.md [hunk 1]
```diff
@@ -22,9 +22,9 @@ aliases: ['/docs-cn/tidb-data-migration/dev/manage-source/']
 MKxn0Qo3m3XOyjCnhEMtsUCm83EhGQDZ/T4=
 ```
 
-## 加载数据源配置
+## 加载、列出、移除数据源
```
Comment (GMHDBJD) [inline zh/manage-source.md:25]: 换个标题，比如数据源操作？

**Review by GMHDBJD — COMMENTED** [review]
rest LGTM

File: en/manage-source.md [hunk 2]
```diff
@@ -108,6 +114,142 @@ get-config source mysql-replica-01
...
+            "result": true,
+            "msg": "",
+            "source": "mysql-replica-01",
+            "worker": "dm-worker-1"
+        }
+    ]
+}
+```
+
+## Change the binding relations of upstream MySQL instances and DM-workers
```
Comment (CharLotteiu) [inline en/manage-source.md:146]:
```suggestion
## Change the bindings between upstream MySQL instances and DM-workers
```

File: en/manage-source.md [hunk 3]
```diff
@@ -108,6 +114,142 @@ get-config source mysql-replica-01
...
+
+You can use the `transfer-source` command change the binding relations of upstream MySQL instances and DM-workers.
```
Comment (CharLotteiu) [inline en/manage-source.md:148]:
```suggestion
You can use the command `transfer-source` command to change the bindings between upstream MySQL instances and DM-workers.
```

File: en/manage-source.md [hunk 4]
```diff
@@ -108,6 +114,142 @@ get-config source mysql-replica-01
...
+Transfers a upstream MySQL/MariaDB source to a free worker.
+Usage:
+  dmctl transfer-source <source-id> <worker-id> [flags]
+Flags:
+  -h, --help   help for transfer-source
+Global Flags:
+  -s, --source strings   MySQL Source ID.
+```
+
+Before changing, DM will check if the unbinding worker is still has running tasks. If so, please [pause tasks](pause-task.md) before the changing, and [resume tasks](resume-task.md) after the changing.
```
Comment (CharLotteiu) [inline en/manage-source.md:166]:
```suggestion
Before transferring, DM checks whether the free worker still has running tasks. If the worker does have running tasks, you need to [pause tasks](pause-task.md) before changing the binding and [resume tasks](resume-task.md) after changing the binding.
```

File: en/manage-source.md [hunk 5]
```diff
@@ -108,6 +114,142 @@ get-config source mysql-replica-01
...
+You can use the `transfer-source` command change the binding relations of upstream MySQL instances and DM-workers.
+
+{{< copyable "" >}}
+
+```bash
+help transfer-source
+```
+
+```
+Transfers a upstream MySQL/MariaDB source to a free worker.
```
Comment (CharLotteiu) [inline en/manage-source.md:157]:
```suggestion
Transfers an upstream MySQL/MariaDB source to a free worker.
```

File: en/manage-source.md [hunk 6]
```diff
@@ -108,6 +114,142 @@ get-config source mysql-replica-01
...
+  -h, --help   help for transfer-source
+Global Flags:
+  -s, --source strings   MySQL Source ID.
+```
+
+Before changing, DM will check if the unbinding worker is still has running tasks. If so, please [pause tasks](pause-task.md) before the changing, and [resume tasks](resume-task.md) after the changing.
+
+### Usage example
+
+If you don't know the binding relation of DM-workers, you could use `dmctl --master-addr <master-addr> list-member --worker` to list current binding relations.
```
Comment (CharLotteiu) [inline en/manage-source.md:170]:
```suggestion
If you do not know the bindings of DM-workers, you can run `dmctl --master-addr <master-addr> list-member --worker` to list the current bindings of all workers.
```

File: en/manage-source.md [hunk 7]
```diff
@@ -108,6 +114,142 @@ get-config source mysql-replica-01
...
+                        "source": ""
+                    }
+                ]
+            }
+        }
+    ]
+}
+```
+
+In this example `mysql-replica-01` is bound to `dm-worker-1`. The below command should transfer the source to bind to `dm-worker-2`.
```
Comment (CharLotteiu) [inline en/manage-source.md:206]:
```suggestion
In the above example, `mysql-replica-01` is bound to `dm-worker-1`. The below command transfers the binding worker of `mysql-replica-01` to `dm-worker-2`.
```

File: en/manage-source.md [hunk 8]
```diff
@@ -108,6 +114,142 @@ get-config source mysql-replica-01
...
+```
+
+```
+{
+    "result": true,
+    "msg": ""
+}
+```
+
+Check the command has taken effect by `dmctl --master-addr <master-addr> list-member --worker`.
```
Comment (CharLotteiu) [inline en/manage-source.md:221]:
```suggestion
Check whether the command takens effect by `dmctl --master-addr <master-addr> list-member --worker`.
```

File: zh/manage-source.md [hunk 9]
```diff
@@ -151,4 +151,118 @@ operate-source show
...
+  dmctl transfer-source <source-id> <worker-id> [flags]
+
+Flags:
+  -h, --help   help for transfer-source
+
+Global Flags:
+  -s, --source strings   MySQL Source ID.
+```
+
+在进行改变之前，DM 会检查旧 worker 是否正在运行同步任务，如果正在运行则需要先[暂停任务](pause-task.md)，并在改变绑定关系后[恢复任务](resume-task.md)。
```
Comment (CharLotteiu) [inline zh/manage-source.md:179]: What does `旧 worker` mean? The free worker?

Comment (lance6716) [reply zh/manage-source.md:179]: the worker that will unbind. Will change description later

File: en/manage-source.md (hunk 4 above)
Comment (lance6716) [reply en/manage-source.md:166]: not "the free worker", will change description later

File: zh/manage-source.md (hunk 9 above)
Comment (lance6716) [reply zh/manage-source.md]: https://github.com/pingcap/docs-dm/pull/605/commits/12cbfef9d525f7b374f0c6bd999a906d2f9415cc

PTAL

File: en/manage-source.md [hunk 10]
```diff
@@ -108,6 +114,142 @@ get-config source mysql-replica-01
...
+Transfers an upstream MySQL/MariaDB source to a free worker.
+Usage:
+  dmctl transfer-source <source-id> <worker-id> [flags]
+Flags:
+  -h, --help   help for transfer-source
+Global Flags:
+  -s, --source strings   MySQL Source ID.
+```
+
+Before transferring, DM checks whether the worker to be unbound still has running tasks. If the worker does have running tasks, you need to [pause tasks](pause-task.md) before changing the binding and [resume tasks](resume-task.md) after changing the binding.
```
[the remark this answers is not in the document]
Comment (lance6716) [reply en/manage-source.md:166]: should we use `[resume the tasks]` with `[pause the tasks]`

[the remark this answers is not in the document]
Comment (yikeke) [reply en/manage-source.md:166]: Yes, adding "the" is better

**Comment by yikeke** [comment]
There are some external link errors:
image

Please fix them in another PR. @CharLotteiu

**Review by yikeke — APPROVED** [review]

**Pull request merged**